packages/material-ui/src/styles/muiStyled.js

```diff
@@ -0,0 +1,62 @@
+import styled from '@material-ui/styled-engine';
+import { propsToClassKey } from '@material-ui/styles';
+import defaultTheme from './defaultTheme';
+
+const getStyleOverrides = (name, theme) => {
+  let styleOverrides = {};
+
+  if (
+    theme &&
+    theme.components &&
+    theme.components[name] &&
+    theme.components[name].styleOverrides
+  ) {
+    styleOverrides = theme.components[name].styleOverrides;
+  }
+
+  return styleOverrides;
+};
+
+const getVariantStyles = (name, theme) => {
+  let variants = [];
+  if (theme && theme.components && theme.components[name] && theme.components[name].variants) {
+    variants = theme.components[name].variants;
+  }
+
+  const variantsStyles = {};
+
+  variants.forEach((definition) => {
+    const key = propsToClassKey(definition.props);
+    variantsStyles[key] = definition.style;
+  });
+
+  return variantsStyles;
+};
+
+const shouldForwardProp = (prop) => prop !== 'state' && prop !== 'theme';
+
+const muiStyled = (el, styledParams, muiConfig) => {
+  const result = styled(el, { shouldForwardProp, ...styledParams });
+  const muiFunc = (...params) => {
+    const name = muiConfig.muiName;
+
+    if (muiConfig.overridesResolver) {
+      params.push((props) => {
+        const theme = props.theme || defaultTheme;
+        return muiConfig.overridesResolver(props, getStyleOverrides(name, theme), name);
+      });
+    }
+
+    if (muiConfig.variantsResolver) {
+      params.push((props) => {
+        const theme = props.theme || defaultTheme;
+        return muiConfig.variantsResolver(props, getVariantStyles(name, theme), theme, name);
+      });
+    }
+
+    return result(...params);
```

Comment:
Same issue here: These are generic terms without any meaning.

Reply:
`styled-components` doesn't work if we don't spread. I had initially `result(params)` and it took me some time to realize why it didn't work 😕

Reply:
Mainly concerned with `result` and `params` naming.

Reply:
Ah sorry was confused that you talked about the spreading. Will improve the names in the whole method